applications/OptimizationApplication/python_scripts/utilities/opt_projection.py

```diff
@@ -0,0 +1,190 @@
+from typing import Optional
+from abc import ABC, abstractmethod
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_problem import OptimizationProblem
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import ContainerExpressionTypes
+
+class Projection(ABC):
+    """Projection methods to convert given x space values to given y space values
```

Comment:
Projection is too general name for class name and file. What about DesignVariableProjection or DesignVariableProjectionMethod ???

Reply:
done :)